pkg/sse/stream.go

```diff
@@ -0,0 +1,74 @@
+package sse
+
+import (
+	"encoding/json"
+	"net/http"
+	"reflect"
+
+	hh "github.com/flynn/flynn/pkg/httphelper"
+)
+
+type Stream struct {
+	w       *Writer
+	done    chan struct{}
+	cleanup func()
+	closed  bool
+}
+
+func ServeStream(w http.ResponseWriter, ch interface{}, cleanup func()) *Stream {
```

Comment:
Why the cleanup function? It'd be better if `Close` blocked until the stream sender is done.